app/controllers/concerns/ial2_profile_concern.rb

```diff
@@ -1,9 +1,22 @@
 module Ial2ProfileConcern
   extend ActiveSupport::Concern
 
-  def cache_active_profile(raw_password)
+  def cache_profiles(raw_password)
+    pending_profile = current_user.pending_profile
+    if pending_profile.present?
```

Comment:
I am considering wrapping this in a block to only run it if the `write` flag is enabled.

Reply:
Actually that does not work because then the pending profile will never get read 🤔